controllers/utils/manifestworkreplicaset.go

```diff
@@ -121,7 +122,71 @@ func GeneratePrepManifestWorkReplicaset(name, namespace string, ibu *lcav1alpha1
 	return generateManifestWorkReplicaset(name, namespace, expectedValueAnn, jsonPaths, ibu)
 }
 
-func generateManifestWorkReplicaset(name, namespace, expectedValueAnn string, jsonPaths []mwv1.JsonPath, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+// GeneratePermissionsManifestWorkReplicaset returns a ManifestWorkReplicaset for permissions required for work agent to access IBU
+func GeneratePermissionsManifestWorkReplicaset(name, namespace string) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+	clusterRole := &rbac.ClusterRole{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "open-cluster-management:klusterlet-work:ibu-role",
+			Labels: map[string]string{
+				"open-cluster-management.io/aggregate-to-work": "true",
+			},
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{"lca.openshift.io"},
+				Resources: []string{"imagebasedupgrades"},
+				Verbs:     []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+			},
+		},
+	}
+	crBytes, err := clusterRoleToBytes(clusterRole)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert ClusterRole to []bytes: %w", err)
+	}
+	manifestConfigs := []mwv1.ManifestConfigOption{}
+	mwrs := &mwv1alpha1.ManifestWorkReplicaSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: mwv1alpha1.ManifestWorkReplicaSetSpec{
+			PlacementRefs: []mwv1alpha1.LocalPlacementReference{
+				{
+					Name: "dummy",
+				},
+			},
+			ManifestWorkTemplate: mwv1.ManifestWorkSpec{
+				Workload: mwv1.ManifestsTemplate{
+					Manifests: []mwv1.Manifest{{RawExtension: runtime.RawExtension{Raw: crBytes}}},
+				},
+				DeleteOption:    &mwv1.DeleteOption{PropagationPolicy: mwv1.DeletePropagationPolicyTypeOrphan},
+				ManifestConfigs: manifestConfigs,
+			},
+		},
+	}
+	return mwrs, nil
+}
+
+func clusterRoleToBytes(cr *rbac.ClusterRole) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	rbac.AddToScheme(scheme)
+	s := serializer.NewSerializerWithOptions(serializer.DefaultMetaFactory, scheme, scheme, serializer.SerializerOptions{})
+	gvks, isUnversioned, err := scheme.ObjectKinds(cr)
+	if err != nil {
+		return []byte{}, err
+	}
+	if !isUnversioned && len(gvks) == 1 {
+		cr.TypeMeta = metav1.TypeMeta{
+			Kind:       gvks[0].Kind,
+			APIVersion: gvks[0].GroupVersion().Identifier(),
+		}
+	}
+	var dst bytes.Buffer
+	s.Encode(cr, &dst)
+	return dst.Bytes(), nil
```

Comment:
Ah surprised that upstream doesn't provided a simple byte conversion func. Seems like super useful use case. Nothing in the Unstructured Objects lib either yeah?

Reply:
How about something like this https://github.com/open-cluster-management-io/api/blob/bd6f2229b53cb61c76ebf528c73bc925d8e0f4c8/utils/work/v1/workapplier/workapplier.go#L166-L181 ?

Reply:
this does not work. It doesn't include apiversion, group and kind.

Reply:
Good to know! Thanks for checking